apps/src/dance/ai/DanceAiModalFlipCard.tsx

```diff
@@ -0,0 +1,129 @@
+import React, {useMemo} from 'react';
+
+import moduleStyles from './dance-ai-modal.module.scss';
+import classNames from 'classnames';
+import AiVisualizationPreview from './AiVisualizationPreview';
+import AiBlockPreview from './AiBlockPreview';
+
+import {DanceAiModalMode, DanceAiPreviewButtonToggleState} from './types';
+import {getPreviewCode} from './utils';
+
+type DanceAiModalFlipCardProps = {
+  mode: DanceAiModalMode;
+  generatingProgressStep: number;
+  badGeneratedResultsCount: number;
+  currentToggle: DanceAiPreviewButtonToggleState;
+  previewAppearDuration: number;
+  currentGeneratedEffect: any;
```

Comment:
Is there a reason you didn't use `GeneratedEffect` type here and below?

Reply:
Nope, just an oversight. Thanks for the catch!